hoomd/hpmc/UpdaterMuVT.h

```diff
@@ -482,6 +556,270 @@ bool UpdaterMuVT<Shape>::boxResizeAndScale(unsigned int timestep, const BoxDim o
+                        {
+                        if (detail::overlap(aabb_tree.getNodeAABB(cur_node_idx), aabb))
+                            {
+                            if (aabb_tree.isNodeLeaf(cur_node_idx))
+                                {
+                                for (unsigned int cur_p = 0; cur_p < aabb_tree.getNodeNumParticles(cur_node_idx); cur_p++)
+                                    {
+                                    // read in its position and orientation
+                                    unsigned int j = aabb_tree.getNodeParticle(cur_node_idx, cur_p);
+
+                                    Scalar4 postype_j;
+                                    Scalar4 orientation_j;
+
+                                    // load the old position and orientation of the j particle
+                                    postype_j = h_postype.data[j];
+                                    orientation_j = h_orientation.data[j];
+
+                                    // compute the particle position scaled in the old box
+                                    f = new_box.makeFraction(make_scalar3(postype_j.x,postype_j.y,postype_j.z));
+                                    vec3<Scalar> pos_j_old(old_box.makeCoordinates(f));
+
+                                    // put particles in coordinate system of particle i
+                                    vec3<Scalar> r_ij = pos_j_old - pos_test_image_old;
+
+                                    unsigned int typ_j = __scalar_as_int(postype_j.w);
+                                    Shape shape_j(quat<Scalar>(orientation_j), params[typ_j]);
+
+                                    if (h_overlaps.data[overlap_idx(typ_j, type_d)]
+                                        && check_circumsphere_overlap(r_ij, shape_test, shape_j)
+                                        && test_overlap(r_ij, shape_test, shape_j, err_count))
+                                        {
+                                        overlap = true;
+
+                                        // depletant is ignored for any overlap in the old configuration
+                                        overlap_old = true;
+                                        break;
+                                        }
+                                    }
+                                }
+                            }
+                        else
+                            {
+                            // skip ahead
+                            cur_node_idx += aabb_tree.getNodeSkip(cur_node_idx);
+                            }
+                        if (overlap)
+                            break;
+                        }  // end loop over AABB nodes
+
+                    if (overlap)
+                        break;
+
+                    } // end loop over images
+
+                if (!overlap)
+                    {
+                    // depletant in free volume
+                    extra_ndof++;
```

Comment:
Why does a depletant only add 1 DOF, not 3? What happens in quermass mode when the depletants are anisotropic and have rotational degrees of freedom? I don't think I'm quite understanding what this number of measuring, I see that it's used in the Boltzmann factor calculation in the `update` method but I don't quite follow the logic.

Reply:
It's used for the box resize acceptance criterion, therefore 1 depletant == 1 particle degree of freedom. We should revisit this because it assumes grand-canonical statistics, with depletants being true (but noninteracting) particles that count towards the pressure. I believe we should introduce an option to make depletants only provide the multi body pair potential but not a pressure contribution. This would then also enable NPT depletants in a single box.

Reply:
Got it, I'll follow up on the issue that you made.